src/Workspaces/CSharp/Portable/LanguageServices/CSharpSemanticFactsService.cs

```diff
@@ -227,6 +227,38 @@ public ForEachSymbols GetForEachSymbols(SemanticModel semanticModel, SyntaxNode
             }
         }
 
+        public ImmutableArray<IMethodSymbol> GetDeconstructionSymbols(SemanticModel semanticModel, SyntaxNode node)
```

Comment:
maybe GetDeconstructionMethods?

Reply:
Also... on a personal level... i don't like having a single helper method that works on all these types.  I think i'd prefer a GetAssignmentDeconstructionSymbols and GetForEachStatementDeconstructionSymbols.